tck/src/main/java/ee/jakarta/tck/concurrent/spec/Platform/dd/DeploymentDescriptorServlet.java

```diff
@@ -45,13 +52,146 @@
+    
+    // Managed Executor Services
+    @Inject
+    private ManagedExecutorService injectedDefMES;
+
+    @Inject
+    @CustomQualifier2
+    private ManagedExecutorService injectedMESD;
+
+    @Inject
+    @CustomQualifier2
+    @InvalidQualifier3
+    private ManagedExecutorService notInjectedMESD;
+    
+    // Managed Scheduled Executor Services
+    @Inject
+    private ManagedExecutorService injectedDefMSES;
+
+    @Inject
+    @CustomQualifier1
+    @CustomQualifier2
+    private ManagedExecutorService injectedMSESD;
+
+    @Inject
+    @CustomQualifier1
+    private ManagedExecutorService notInjectedMSESD;
+    
+    // Managed Thread Factory
+    @Inject
+    private ManagedThreadFactory injectedDefMTF;
+
+    @Resource(lookup = "java:app/concurrent/ThreadFactoryD")
+    private ManagedThreadFactory resourceMTFD;
+
+    @Inject
+    @CustomQualifier1
+    private ManagedThreadFactory notInjectedMTFD;
+    
+    private Integer executeCallableWithContext(final ContextService svc, final int value) throws Exception {
+        try {
+            IntContext.set(value);
+            Callable<Integer> result = svc.contextualCallable(() -> {
+                return IntContext.get();
+            });
+            return result.call();
+        } finally {
+            IntContext.set(0);
+        }
+    }
+
+    public void testDeploymentDescriptorDefinesQualifiers() throws Throwable {
+        assertAll("Context Service Tests",
+                () -> assertNotNull(injectedDefContextSvc,
+                        "Default contextService was not injected when no qualifiers were present."),
+                () -> assertNotNull(injectedContextD,
+                        "Deployment Descriptor defined contextService was not inject with valid qualifier."),
+                () -> assertNull(notInjectedContextD,
+                        "A contextService was injected with a qualifier which was not defined in it's deployment description"));
+        
+        //  Verify injected and lookup default context service are the same
+        ContextService lookupDefContextSvc = InitialContext.doLookup("java:comp/DefaultContextService");
+        
+        Integer expected1 = executeCallableWithContext(lookupDefContextSvc, 95);
+        Integer actual1 = executeCallableWithContext(injectedDefContextSvc, 95);
+        
+        assertEquals(expected1, actual1, "Default Context Service behavior differed between injection and lookup");
+        
+        //  Verify injected and lookup deployment descriptor defined context service are the same
+        ContextService lookupContextD = InitialContext.doLookup("java:global/concurrent/ContextD");
+        
+        Integer actual2 = executeCallableWithContext(lookupContextD, 65);
+        Integer expected2 = executeCallableWithContext(injectedContextD, 85);
+        
+        assertEquals(actual2, expected2, "Deployment descriptor defined Context Service behavior differed between injection and lookup");
```

Comment:
This works, although expected and actual seem backwards and it seems unnecessarily complicated (as well as permissive) to compute what the expected value ought to be by trying it out on another instance obtained a different way that is presumed to be working rather than just asserting the expected value directly.

```suggestion
        assertEquals(Integer.valueOf(0), executeCallableWithContext(lookupContextD, 65), "Deployment descriptor defined Context Service that was looked up did not clear the IntContext as configured.");
        assertEquals(Integer.valueOf(0), executeCallableWithContext(injectedContextD, 85), "Deployment descriptor defined Context Service that was injected based on a qualifier did not clear the IntContext as configured.");
```